### PR TITLE
Fix for Client does not support keys containing "spaces"

### PR DESCRIPTION
**Pull request by jkobus**
refs #19

I wanted to add some unit tests but mocking Translation and Asset takes a lot of time and does not look pretty, as things are tightly connected to each other. That would need some care in the near future.

### CODE REVIEW

**Comment by jkobus** [comment]
There is also a problem with "!" sign, as asset is beeing  created, but api wont accept translation for it: "Asset not in project".

File: src/Api/Asset.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public function create(string $projectKey, string $id)
     {
         $param = [
             'name' => $id,
-            'id' => $id,
+            'id' => urlencode($id),
```
Comment (Nyholm) [inline src/Api/Asset.php]: The $param will go through a `http_build_query`. Is This really needed?

Comment (jkobus) [reply src/Api/Asset.php]: Checking.

**Comment by Nyholm** [comment]
I hope it is okey if I ping @timwhitlock. =)

The issue here is that we have spaces the the asset key. 
So with the asset key: "some translation key"
The URL will be: `/api/translations/some translation key/pl?key=12`

@jkobus solve it with `urlencode` (see this PR) but there is still a problem with exclamation marks. (https://github.com/FriendsOfApi/localise.biz/pull/20#issuecomment-316060082)

Tim: Can you give us some guidance?

**Comment by jkobus** [comment]
1. POST https://localise.biz/api/assets?key=xxx
`name=Sign+in%21&id=Sign+in%21&type=text&default=untranslated`
OK

2. POST-https://localise.biz/api/translations/Sign+in%21/pl?key=xxx
` ` (empty body, as there is not translation)

API responds with: `{"status":404,"error":"Asset not in project"}`

File: src/Api/Asset.php (hunk 1 above)
Comment (jkobus) [reply src/Api/Asset.php]: Not needed indeed.

**Comment by timwhitlock** [comment]
The API expects `%20` for spaces. (RFC 3986) `+` characters are passed though literally.

PHP clients should use `rawurlencode` instead of `urlencode`.

**Comment by Nyholm** [comment]
Thank you. That is no issue. It's the exclamation marks. See https://github.com/FriendsOfApi/localise.biz/pull/20#issuecomment-316064316

**Comment by jkobus** [comment]
Actually after changing it to rawurlencode the string "Sign in!" works and was accepted by the API. PR was updated.

I ran "send" for all legacy translations and it looks that this fix worked.

**Comment by Nyholm** [comment]
Okey  good. 

Thank you both!

**Pull request merged**